### PR TITLE
Add ngtcp2 to oqs-demos

### PR DESCRIPTION
**Pull request by Keelan10**
Hi, I integrated OQS-OpenSSL-QUIC into ngtcp2 and provided Dockerfiles to fully automate the integration.

Thanks for reviewing!

### CODE REVIEW

File: ngtcp2/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+The following commands
+
+```
+docker build -t ngtcp2_client -f Dockerfile-client .
+docker run -it ngtcp2_client bash
+```
+
+will build up and run the container for the QSC-enabled ngtcp2 client.
+
+Ensure that the server and the client run on the same network.
```
Comment (baentsch) [inline ngtcp2/README.md:52]: Suggest to add code sample using `docker network` like https://github.com/open-quantum-safe/oqs-demos/blob/main/haproxy/USAGE.md

**Review by baentsch — CHANGES_REQUESTED** [review]
Thanks very much for this contribution. This already looks great. Are you interested that this is published on https://hub.docker.com/orgs/openquantumsafe/repositories ? If so, there's two additions needed:
1) A USAGE.md file that could be uploaded as-is to docker hub as documentation for use of the built image. This basically means a split of your current README into a (build-)README and a USAGE.md file. Please see examples in the other demos; of particular relevance might be https://github.com/open-quantum-safe/oqs-demos/blob/main/haproxy/USAGE.md as it also documents docker network issues. 
2) If OK  for you, please also add a suitable entry to the CI system (e.g., along the lines of [the oqs-quic build](https://github.com/open-quantum-safe/oqs-demos/blob/7966f93beae30974872b4321aa290d4d0b882f58/.circleci/config.yml#L102) ).

If you're unfamilar with this, no problem -- I'll then do a followup PR; but please do let us know whether you're OK with this being on docker hub, though.

In any case, please add suitable entries to "package" and "Contributors" to the main [oqs-demos README](https://github.com/open-quantum-safe/oqs-demos/blob/main/README.md).

**Comment by Keelan10** [comment]
Having this published on docker hub would be great. I am okay with that idea.
I would be grateful if you could add an entry to the CI system.

Thanks

**Review by baentsch — APPROVED** [review]
LGTM. Thanks for the changes. Will add CI PR.

**Comment by baentsch** [comment]
@Keelan10 While doing the CI integration I hit a problem: The CI-pass condition is to start client and server images automatically and successfully establish a connection. To do that what normally works is assigning the server image a name that then becomes the DNS name under which one can connect. For the client image this works: A server started with `docker run --network ngtcp2-test --name ngtcp2server -it ngtcp-server bash` can be correctly reached from within the client image with the document command `client ngtcp2server 6000`. 

However, I cannot start the server with the equivalent command `server ngtcp2server 6000 CA.key CA.crt` but _have_ to determine its IP address first and then call `server 172.18.0.2 6000 CA.key CA.crt`... -> Is there no way to use the IP address of the server when starting it? The error message I get is 
```
root@31622fcd3db7:~# server ngtcp2server 6000 CA.key CA.crt 
Using document root /root/
getaddrinfo: No address associated with hostname
```

This is odd as `ping ngtcp2server` works just fine:
```
root@31622fcd3db7:~# ping ngtcp2server                    
PING ngtcp2server (172.18.0.2) 56(84) bytes of data.
64 bytes from 31622fcd3db7 (172.18.0.2): icmp_seq=1 ttl=64 time=0.077 ms
64 bytes from 31622fcd3db7 (172.18.0.2): icmp_seq=2 ttl=64 time=0.092 ms
```

Is this easily explicable to you? I didn't want to debug into ngtcp2, really....

**Comment by Keelan10** [comment]
It looks like you found the fix already. 
Thanks for the review. I will use the insight you provided to simplify the usage documentation.

**Pull request merged**